### PR TITLE
Moved web assets to web/ and implemented a pure PHP compression solution for CSS and JS files

### LINKED ISSUE #5: Why are assets in app/Resources/ ?
This seems a bit strange to me. All assets are already ready compiled and ready to be used. Why not placing them in the `web/` directory directly? (seems duplication for nothing)

Also, IIRC `assets:install` doesn't search through the `app/Resources/assets` directory.

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #5

### CODE REVIEW

File: app/Resources/views/base.html.twig [hunk 1]
```diff
@@ -128,11 +128,11 @@
         {% endblock %}
 
         {% block javascripts %}
-        {# uncomment the following lines to compile JavaScript assets with Assetic
-            {% javascripts output="js/app.js"
-                "%kernel.root_dir%/Resources/assets/js/jquery-2.1.3.min.js"
-                "%kernel.root_dir%/Resources/assets/js/bootstrap-3.3.4.min.js"
-                "%kernel.root_dir%/Resources/assets/js/highlight.pack.js" %}
+        {# uncomment the following lines to combine and minimize JavaScript assets with Assetic
+            {% javascripts filter="jsqueeze" output="js/app.js"
```
Comment (stof) [inline app/Resources/views/base.html.twig]: should be ``?jsqueeze`` to apply the minification only in non-debug mode

File: app/Resources/views/base.html.twig [hunk 2]
```diff
@@ -7,11 +7,11 @@
...
-            {% stylesheets output="css/app.css"
-                "%kernel.root_dir%/Resources/assets/scss/bootstrap.scss"
-                "%kernel.root_dir%/Resources/assets/scss/font-awesome.scss"
-                "%kernel.root_dir%/Resources/assets/css/*.css"
-                "%kernel.root_dir%/Resources/assets/scss/main.scss"
+            {% stylesheets filter="scssphp" output="css/app.css"
+                "%kernel.root_dir%/../web/assets/scss/bootstrap.scss"
+                "%kernel.root_dir%/../web/assets/scss/font-awesome.scss"
+                "%kernel.root_dir%/../web/assets/css/*.css"
+                "%kernel.root_dir%/../web/assets/scss/main.scss"
```
Comment (stof) [inline app/Resources/views/base.html.twig]: can be simpler: ``'assets/scss/main.scss``. Assetic will read relative paths from the web dir

File: app/config/config.yml [hunk 3]
```diff
@@ -44,12 +44,12 @@ assetic:
...
     bundles:        [ ]
-    #java: /usr/bin/java
     filters:
         cssrewrite: ~
-        sass:
-            bin:      "/usr/bin/sass"
-            apply_to: "\.scss$"
+        jsqueeze: ~
+        scssphp:
+            # the formatter must be the FQCN (don't use the 'compressed' value)
```
Comment (stof) [inline app/config/config.yml]: really ? for lessphp, using ``compressed`` works fine.

File: app/Resources/views/base.html.twig (hunk 2 above)
Comment (javiereguiluz) [inline app/Resources/views/base.html.twig]: You are right. Done. Thanks.

File: app/config/config.yml (hunk 3 above)
Comment (javiereguiluz) [inline app/config/config.yml]: If I use the following config:

```yaml
assetic:
    filters:
        scssphp:
            formatter: 'compressed'
```

I get the following error:

assetic_error

Comment (stof) [inline app/config/config.yml]: @javiereguiluz can you give the stack trace ?

Comment (javiereguiluz) [inline app/config/config.yml]: This is the full log information:

```
[2015-04-10 12:19:41] request.INFO: Matched route "homepage" (parameters: "_controller": "Symfony\Bundle\FrameworkBundle\Controller\TemplateController::templateAction", "template": "default/homepage.html.twig", "_route": "homepage") [] []
[2015-04-10 12:19:41] request.INFO: Matched route "_assetic_5c80c23_0" (parameters: "_controller": "assetic.controller:render", "name": "5c80c23", "pos": "0", "_format": "css", "_route": "_assetic_5c80c23_0") [] []
[2015-04-10 12:19:42] request.INFO: Matched route "_assetic_5c80c23_1" (parameters: "_controller": "assetic.controller:render", "name": "5c80c23", "pos": "1", "_format": "css", "_route": "_assetic_5c80c23_1") [] []
[2015-04-10 12:19:42] request.INFO: Matched route "_assetic_5c80c23" (parameters: "_controller": "assetic.controller:render", "name": "5c80c23", "pos": "null", "_format": "css", "_route": "_assetic_5c80c23") [] []
[2015-04-10 12:19:42] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ClassNotFoundException: "Attempted to load class "compressed" from the global namespace. Did you forget a "use" statement?" at /Users/javier/sf/symfony-demo/vendor/leafo/scssphp/src/Compiler.php line 129 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ClassNotFoundException(code: 0): Attempted to load class \"compressed\" from the global namespace.\nDid you forget a \"use\" statement? at /Users/javier/sf/symfony-demo/vendor/leafo/scssphp/src/Compiler.php:129)"} []
[2015-04-10 12:19:42] request.INFO: Matched route "_assetic_5c80c23_2" (parameters: "_controller": "assetic.controller:render", "name": "5c80c23", "pos": "2", "_format": "css", "_route": "_assetic_5c80c23_2") [] []
[2015-04-10 12:19:43] request.INFO: Matched route "_assetic_5c80c23_3" (parameters: "_controller": "assetic.controller:render", "name": "5c80c23", "pos": "3", "_format": "css", "_route": "_assetic_5c80c23_3") [] []
[2015-04-10 12:19:43] request.INFO: Matched route "_assetic_5c80c23_4" (parameters: "_controller": "assetic.controller:render", "name": "5c80c23", "pos": "4", "_format": "css", "_route": "_assetic_5c80c23_4") [] []
[2015-04-10 12:19:43] request.INFO: Matched route "_wdt" (parameters: "_controller": "web_profiler.controller.profiler:toolbarAction", "token": "8282c1", "_route": "_wdt") [] []
```

Comment (stof) [inline app/config/config.yml]: What I want is the stack trace of the exception, not the logs (run your assetic dump command in verbose mode to get the stacktrace)

Comment (javiereguiluz) [inline app/config/config.yml]: Sorry! Here it is:

```
$ prod a:d -vvv
Dumping all prod assets.
Debug mode is off.

14:55:25 [file+] /Users/javier/sf/symfony-demo/app/../web/css/app.css
        /Users/javier/sf/symfony-demo/app/../web/assets/scss/bootstrap.scss
        /Users/javier/sf/symfony-demo/app/../web/assets/scss/font-awesome.scss
        /Users/javier/sf/symfony-demo/app/../web/assets/css/font-lato.css
        /Users/javier/sf/symfony-demo/app/../web/assets/css/highlight-solarized-light.css
        /Users/javier/sf/symfony-demo/app/../web/assets/scss/main.scss
[2015-04-10 14:55:25] php.EMERGENCY: Fatal Error: Class 'compressed' not found {"type":1,"file":"/Users/javier/sf/symfony-demo/vendor/leafo/scssphp/src/Compiler.php","line":129,"level":0,"stack":[]}



  [Symfony\Component\Debug\Exception\ClassNotFoundException]
  Attempted to load class "compressed" from the global namespace.
  Did you forget a "use" statement?



Exception trace:
 () at /Users/javier/sf/symfony-demo/vendor/leafo/scssphp/src/Compiler.php:129

```

Comment (stof) [inline app/config/config.yml]: indeed. the ScssPhp library expects a class name. It is always nice to see consistency between libraries created by the same dev :smile:

**Comment by javiereguiluz** [comment]
I'm not sure about merging this change. I'd love to hear first the opinions of @weaverrryan and @WouterJ about it. The both know this topic well and they seem to have strong opinions about it (and definitely more qualified than mine!)

**Comment by stof** [comment]
@weaverryan ping as there was a typo when mentioning you previously

@WouterJ any opinion on this ?

**Comment by WouterJ** [comment]
* I don't like depending on ports, they never are as good as the original ones. If it works with the code in the lib, I'm fine with using it here (although I don't believe we should recommend it).
* Imo, sass/scss files are not public files, they shouldn't be accessed. That's why I vote to put these in `app/Resources/public` or the like, at least outside of `web/`. This is also following what the best practices guide says.

**Comment by javiereguiluz** [comment]
This PR is ready for the final review:

  * The assets' sources are moved back to `app/Resources/assets/`
  * The compiled assets are `web/css/app.css` and `web/js/app.js`
  * The fonts have been moved to `web/fonts/` to not duplicate them
  * The SCSS is now compiled and the CSS are minified with a PHP library (`Leafo\ScssPhp`)
  * The JS are minified with a PHP library (JsSqueeze)

The result is that:

  * The application still looks nice without configuring anything or executing any command.
  * If someone uncomments Assetic blocks, everything should still work because we use PHP-only solutions.

**Comment by stof** [comment]
@javiereguiluz it would be great to provide a way to recompile the assets for contributors (using ``assetic:dump`` won't work as the Assetic blocks are commented), and explaining it in the contributing file.

**Comment by javiereguiluz** [comment]
@stof I think that's a very nice idea! Thanks. I've added a note about contributing in 62417ca

**Comment by stof** [comment]
@javiereguiluz I suggest putting this in CONTRIBUTING.md instead, to follow the github conventions

**Comment by javiereguiluz** [comment]
@stof much better indeed. Done in a88b360. Thanks.

**Pull request closed without merging**